js&css/web-accessible/www.youtube.com/player.js

```diff
@@ -488,7 +488,20 @@ ImprovedTube.playerQuality = function (quality = this.storage.player_quality) {
 		player.setPlaybackQuality(quality);
 		player.dataset.defaultQuality = quality;
 	}
+
+    // --- Place your advanced buffering logic here ---
+    const video = this.elements.video;
+    if (ImprovedTube.storage.get('advanced_video_buffering') === true && video) {
+  video.preload = 'auto';
+  let originalTime = video.currentTime;
+  let bufferSeconds = ImprovedTube.storage.get('buffer_seconds') || 30;
+  let bufferTime = Math.min(video.duration || (originalTime + bufferSeconds), originalTime + bufferSeconds);
+  video.currentTime = bufferTime;
+  video.currentTime = originalTime;
+}
```

Comment:
The codebase seems to use tabs for indentation (which I personally approve!).
```suggestion
	if (ImprovedTube.storage.get('advanced_video_buffering') === true && video) {
		video.preload = 'auto';
		let originalTime = video.currentTime;
		let bufferSeconds = ImprovedTube.storage.get('buffer_seconds') || 30;
		let bufferTime = Math.min(video.duration || (originalTime + bufferSeconds), originalTime + bufferSeconds);
		video.currentTime = bufferTime;
		video.currentTime = originalTime;
	}
```